apl-core/src/main/java/com/apollocurrency/aplwallet/apl/alias/service/AliasServiceImpl.java

```diff
@@ -0,0 +1,184 @@
+ * LICENSE.txt file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+/*
+ * Copyright © 2018-2020 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.alias.service;
+
+import com.apollocurrency.aplwallet.apl.alias.converter.IteratorToStreamConverter;
+import com.apollocurrency.aplwallet.apl.alias.dao.AliasOfferTable;
+import com.apollocurrency.aplwallet.apl.alias.dao.AliasTable;
+import com.apollocurrency.aplwallet.apl.alias.entity.Alias;
+import com.apollocurrency.aplwallet.apl.alias.entity.AliasOffer;
+import com.apollocurrency.aplwallet.apl.core.app.Blockchain;
+import com.apollocurrency.aplwallet.apl.core.app.Transaction;
+import com.apollocurrency.aplwallet.apl.core.db.DbClause;
+import com.apollocurrency.aplwallet.apl.core.db.DbIterator;
+import com.apollocurrency.aplwallet.apl.core.transaction.messages.MessagingAliasAssignment;
+import com.apollocurrency.aplwallet.apl.core.transaction.messages.MessagingAliasSell;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.stream.Stream;
+
+@Singleton
+public class AliasServiceImpl implements AliasService {
+    private final AliasTable aliasTable;
+    private final AliasOfferTable offerTable;
+    private final Blockchain blockchain;
+    private final IteratorToStreamConverter<Alias> converter;
+
+    /**
+     * Constrictor for unit tests.
+     *
+     * @param aliasTable
+     * @param offerTable
+     * @param blockchain
+     * @param converter
+     */
+    public AliasServiceImpl(
+        final AliasTable aliasTable,
+        final AliasOfferTable offerTable,
+        final Blockchain blockchain,
+        final IteratorToStreamConverter converter
+    ) {
+        this.aliasTable = aliasTable;
+        this.offerTable = offerTable;
+        this.blockchain = blockchain;
+        this.converter = converter;
+    }
+
+    @Inject
+    public AliasServiceImpl(
+        final AliasTable aliasTable,
+        final AliasOfferTable offerTable,
+        final Blockchain blockchain
+    ) {
+        this.aliasTable = aliasTable;
+        this.offerTable = offerTable;
+        this.blockchain = blockchain;
+        this.converter = new IteratorToStreamConverter<>();
+    }
+
+    @Override
+    public int getCount() {
+        return aliasTable.getCount();
+    }
+
+    @Override
+    public int getAccountAliasCount(long accountId) {
+        return aliasTable.getCount(new DbClause.LongClause("account_id", accountId));
+    }
+
+    @Override
+    public Stream<Alias> getAliasesByOwner(long accountId, int from, int to) {
+        final DbIterator<Alias> aliasIterator = aliasTable.getManyBy(
+            new DbClause.LongClause("account_id", accountId), from, to
+        );
+
+        return converter.convert(aliasIterator);
+    }
+
+    @Override
+    public Alias getAliasByName(String aliasName) {
+        return aliasTable.getBy(new DbClause.StringClause("alias_name_lower", aliasName.toLowerCase()));
```

Comment:
Pls add check input to prevent NPE - Objects.requireNonNull(..)
`aliasName.toLowerCase()`

Reply:
done